java/org/contikios/cooja/Simulation.java

```diff
@@ -150,68 +158,129 @@ public Simulation(Cooja cooja, long seed) {
     this.cooja = cooja;
     randomGenerator = new SafeRandom(this);
     randomSeed = seed;
+    this.simulationThread = new Thread(() -> {
+      boolean isAlive = true;
+      do {
+        boolean isSimulationRunning = false;
+        EventQueue.Pair nextEvent = null;
+        try {
+          while (isAlive) {
+            Runnable cmd;
+            do {
+              cmd = isSimulationRunning ? commandQueue.poll() : commandQueue.take();
+              if (cmd == null) {
+                continue;
+              }
+              if (cmd == START) {
+                isSimulationRunning = true;
+                setRunning(true);
+              } else if (cmd == STOP) {
+                isSimulationRunning = false;
+                setRunning(false);
+              } else if (cmd == QUIT) {
+                isAlive = false;
+                isShutdown = true;
+                isSimulationRunning = false;
+                setRunning(false);
+                // Process no more commands
+                break;
+              } else {
+                cmd.run();
+              }
+            } while (cmd != null);
+
+            if (isSimulationRunning) {
+              /* Handle one simulation event, and update simulation time */
+              nextEvent = eventQueue.popFirst();
+              assert nextEvent != null : "Ran out of events in eventQueue";
+              assert nextEvent.time >= currentSimulationTime : "Event from the past";
+              currentSimulationTime = nextEvent.time;
+              nextEvent.event.execute(currentSimulationTime);
+            }
+          }
+        } catch (RuntimeException e) {
+          if ("MSPSim requested simulation stop".equals(e.getMessage())) {
+            /* XXX Should be*/
+            logger.info("Simulation stopped due to MSPSim breakpoint");
+          } else {
+            logger.fatal("Simulation stopped due to error: " + e.getMessage(), e);
+            if (!Cooja.isVisualized()) {
+              /* Quit simulator if in test mode */
+              System.exit(1);
+            } else {
+              String title = "Simulation error";
+              if (nextEvent != null && nextEvent.event instanceof MoteTimeEvent moteTimeEvent) {
```

Comment:
I think this part of the change can be merged to master if you put it in a separate PR, that would make this PR a pure whitespace-change for this part of the patch.

Reply:
Done.